File: OpenVehicleApp/src/main/java/com/openvehicles/OVMS/ui/witdet/ReversedSeekBar.java
package com.openvehicles.OVMS.ui.witdet;

import android.content.Context;
import android.graphics.Canvas;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.support.v7.widget.AppCompatSeekBar;

public class ReversedSeekBar extends AppCompatSeekBar {
	public boolean isReversed = true;

	public ReversedSeekBar(Context paramContext, AttributeSet paramAttributeSet) {
		super(paramContext, paramAttributeSet);
	}

	protected void onDraw(Canvas paramCanvas) {
		if (isReversed) {
			paramCanvas.scale(-1.0F, 1.0F, getWidth() / 2.0F, getHeight() / 2.0F);
		}
		try {
			super.onDraw(paramCanvas);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	public boolean onTouchEvent(MotionEvent paramMotionEvent) {
		if (isReversed) {
			paramMotionEvent.setLocation(getWidth() - paramMotionEvent.getX(),
					paramMotionEvent.getY());
		}
		return super.onTouchEvent(paramMotionEvent);
	}
}

File: app/src/main/java/com/openvehicles/OVMS/ui/witdet/ReversedSeekBar.kt
package com.openvehicles.OVMS.ui.witdet

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.appcompat.widget.AppCompatSeekBar

class ReversedSeekBar(
    paramContext: Context?,
    paramAttributeSet: AttributeSet?
) : AppCompatSeekBar(paramContext!!, paramAttributeSet) {

    private var isReversed = true

    override fun onDraw(canvas: Canvas) {
        if (isReversed) {
            canvas.scale(-1.0f, 1.0f, width / 2.0f, height / 2.0f)
        }

        try {
            super.onDraw(canvas)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onTouchEvent(motionEvent: MotionEvent): Boolean {
        if (isReversed) {
            motionEvent.setLocation(
                width - motionEvent.x,
                motionEvent.y
            )
        }

        return super.onTouchEvent(motionEvent)
    }
}